File: internal/commands/changelog.go
package commands

import (
	"fmt"
	"os"

	"github.com/kgsaran/trackfw/internal/changelog"
	"github.com/spf13/cobra"
)

func newChangelogCmd() *cobra.Command {
	var versionFlag string
	var allFlag bool
	cmd := &cobra.Command{
		Use:   "changelog",
		Short: "Show entries from CHANGELOG.md",
		RunE: func(cmd *cobra.Command, args []string) error {
			root, err := os.Getwd()
			if err != nil {
				return err
			}
			content, err := changelog.Read(root)
			if err != nil {
				return err
			}
			if allFlag {
				fmt.Print(content)
				return nil
			}
			sections, err := changelog.ParseSections(content)
			if err != nil {
				return err
			}
			var section changelog.Section
			if versionFlag != "" {
				section, err = changelog.FindVersion(sections, versionFlag)
			} else {
				section, err = changelog.FirstSection(sections)
			}
			if err != nil {
				return err
			}
			fmt.Print(changelog.FormatSection(section))
			return nil
		},
	}
	cmd.Flags().StringVar(&versionFlag, "version", "", "Show a specific version section")
	cmd.Flags().BoolVar(&allFlag, "all", false, "Show the entire CHANGELOG.md")
	return cmd
}

File: npm/src/commands/changelog.js
'use strict'
const { Command } = require('commander')
const { read, parseSections, firstSection, findVersion, formatSection } = require('../changelog')

const cmd = new Command('changelog')
cmd.description('Show entries from CHANGELOG.md')
// enablePositionalOptions escopa a flag local "--version" a este subcomando,
// evitando que o root a intercepte como a flag global "trackfw --version".
cmd.enablePositionalOptions()
cmd.option('--version <x.y.z>', 'Show a specific version section')
cmd.option('--all', 'Show the entire CHANGELOG.md')
cmd.action(async (opts) => {
  try {
    const root = process.cwd()
    const content = read(root)
    if (opts.all) {
      process.stdout.write(content)
      return
    }
    const sections = parseSections(content)
    const section = opts.version ? findVersion(sections, opts.version) : firstSection(sections)
    process.stdout.write(formatSection(section))
  } catch (err) {
    console.error(`Error: ${err.message}`)
    process.exitCode = 1
  }
})

module.exports = cmd
